streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -117,6 +120,19 @@ void handleRebalanceComplete() {
         rebalanceInProgress = false;
     }
 
+    void handleCorruption(final Set<TaskId> taskIds) {
+        for (final TaskId taskId : taskIds) {
+            final Task task = tasks.get(taskId);
+
+            // this call is idempotent so even if the task is only CREATED we can still call it
+            changelogReader.remove(task.changelogPartitions());
```

Comment:
We have to call changelog.remove() before task.closeDirty since now we clear the `stores` map in closeDirty, and after that task.changelogPartitions() would return nothing.